omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,245 @@
+			description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		} else if (rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("identifierType");
+			description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+			description.addSelfLink();
+		}
+		return description;
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType() + " - " + autoGenerationOption.getSource() + " - "
+		        + autoGenerationOption.getLocation();
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("identifierType");
+		description.addProperty("location");
+		description.addRequiredProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = getCreatableProperties();
+		description.removeProperty("identifierType");
+		return description;
+	}
+	
+	@Override
+	public SimpleObject create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		Object manualEntryEnabled = (postBody.get("manualEntryEnabled"));
+		Object automaticallyEnabled = postBody.get("automaticGenerationEnabled");
+		Object sourceUuid = postBody.get("source");
+		Object locationUuid = postBody.get("location");
+		Object identifierTypeUuid = postBody.get("identifierType");
+		ArrayList<String> errors = new ArrayList<String>();
+		if (sourceUuid == null || StringUtils.isBlank(sourceUuid.toString())) {
+			errors.add("source");
+		}
+		if (identifierTypeUuid == null || StringUtils.isBlank(identifierTypeUuid.toString())) {
+			errors.add("identifierType");
+		}
+		if (errors.size() > 0) {
+			throw new ValidationException("The values of the following inputs are missing: " + errors.toString());
+		}
+		IdentifierSource identifierSource = identifierSourceService.getIdentifierSourceByUuid(sourceUuid.toString());
+		if (identifierSource == null) {
+			errors.add("source");
+		}
+		PatientIdentifierType patientIdentifierType = Context.getPatientService()
+		        .getPatientIdentifierTypeByUuid(identifierTypeUuid.toString());
+		if (patientIdentifierType == null) {
+			errors.add("IdentifierType");
+		}
+		if (errors.size() > 0) {
```

Comment:
Is it necessary to repeat this?

Reply:
the first  "if (errors.size() > 0)"  in line 118 handles the case where the user did not provide the IdentifierType Uuid or the Source Uuid. the second one in line 130 handles the case where the IdentifierType Uuid or the Source Uuid provided by the user does not correspond to any identifierType or Source in the system eg  Context.getPatientService().getPatientIdentifierTypeByUuid(identifierTypeUuid.toString()) for such will return a null

Reply:
Have the errors.size check after you have checked for all nulls.
To avoid null pointer exceptions, use the else block to deal with the non null variable.